### PR TITLE
Updated GB postcode regex to assert start and end positions for middle a...

### PR DESCRIPTION
**Pull request by SorX14**
...nd end parts

### CODE REVIEW

**Comment by weierophinney** [comment]
Linking to #7250

**Comment by weierophinney** [comment]
@SorX14 Please provide unit tests that cover the formats the regex now enables.

**Comment by SorX14** [comment]
Just added some tests, which should cover the original set of issues that I ran into and - for the first time - the UK validator is being tested :)

File: tests/ZendTest/I18n/Validator/PostCodeTest.php [hunk 1]
```diff
@@ -35,6 +35,33 @@ public function setUp()
...
+	{
+		$uk_validator = new PostCodeValidator(array('locale' => 'en_GB'));
+		$this->assertEquals($expected, $uk_validator->isValid($postCode));
+	}
+
+	public function UKPostCodesDataProvider()
+	{
+		return array (
+			array ('CA3 5JQ', true),
+			array ('GL15 2GB', true),
```
Comment (malukenho) [inline tests/ZendTest/I18n/Validator/PostCodeTest.php]: Do not use tabs for indentation.

**Comment by weierophinney** [comment]
Merged to develop for release with 2.4.

**Pull request closed without merging**